### PR TITLE
Add TNLRv3 fp16 pattern to Layer Norm fusion

### PR DESCRIPTION
**Pull request by ashbhandare**
Add the following pattern found in TNLR_v3 +Deepspeed fp16 graph:
image

### CODE REVIEW

File: onnxruntime/core/optimizer/layer_norm_fusion.cc [hunk 1]
```diff
@@ -194,24 +201,25 @@ Status LayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
     }
     nodes_to_remove.push_back(pow_node);
 
-    // check if Cast node exists between sub and pow
+    // check if Cast node exists: either between sub and pow, or as second input to pow
     const Node* p_cast_node = graph_utils::FirstParentByType(pow_node, "Cast");
     if (p_cast_node != nullptr) {
       Node& cast_node = *graph.GetNode(pow_node.InputNodesBegin()->Index());
       if (!graph_utils::IsSupportedOptypeVersionAndDomain(cast_node, "Cast", {9, 13}) ||
           cast_node.GetExecutionProviderType() != cast_node.GetExecutionProviderType() ||
```
Comment (ytaous) [inline onnxruntime/core/optimizer/layer_norm_fusion.cc:209]: >cast_node [](start = 50, length = 9)

should be reduce_mean_node - happen to notice this when working on the other fusion

**Comment by ytaous** [comment]
Do we see the same pattern when using ORTTrainer?

**Comment by ytaous** [comment]
Add an UT?

**Comment by ashbhandare** [comment]
> Do we see the same pattern with ORTTrainer?

We should not, as this is due to the export of fp16 graph transformed by deepspeed

File: onnxruntime/core/optimizer/layer_norm_fusion.cc (hunk 1 above)
Comment (ashbhandare) [reply onnxruntime/core/optimizer/layer_norm_fusion.cc:209]: I am not sure if that's right. The Casts here can either be between sub and pow or as second input to Pow. Will sync offline.

**Review by SherlockNoMad — APPROVED** [review]

**Pull request merged**